.github/workflows/build.yml

```
@@ -0,0 +1,132 @@
+name: build-and-deployment
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+
+jobs:
+  deploy-staging:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/develop'
+    steps:
+    - uses: 'actions/checkout@v4'
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v2'
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+    - name: Debug GCP credentials
```

This is not a debugging step @Sachinbisht27. We are copying the content from Github Environment variable to a new file `credentials.json` and using it for connecting with GCP in a few next steps.

```suggestion
    - name: Create credentials.json file using Github Environment variable
```

---

Same we are doing in the Cloud SDK step so we can remove this step also.